lib/dictBuilder/cover.c

```diff
@@ -929,6 +929,65 @@ void COVER_best_finish(COVER_best_t *best, size_t compressedSize,
   }
 }
 
+static void COVER_selectDict(void* dict, size_t* dictBufferCapacity, const void* customDictContent, size_t dictContentSize,
```

Comment:
Instead of using out-parameters, lets return a struct. We also need to handle returning errors.

```c
struct COVER_dictSelection {
  const void* dictContent;
  size_t dictSize;
  size_t totalCompressedSize;
};

COVER_dictSelection COVER_dictSelectionError(size_t error) {
    const COVER_dictSelection selection = { NULL, 0, error };
    return selection;
}

bool COVER_dictSelectionIsError(COVER_dictSelection selection) {
  return ZSTD_isError(selection.totalCompressedSize);
}
```